src/main/java/io/reactivesocket/ReactiveSocket.java

```diff
@@ -95,6 +95,13 @@ public void error(Throwable e) {
     void onRequestReady(Completable c);
 
     /**
+     * Invoked when ReactiveSocket is closed
+     *
+     * @throws IllegalStateException if this method is called more than once per instance
+     */
+    void onShutdown(Completable c);
```

Comment:
Instead of returning void, I would return the previous Completable (and allow changing the callback).